cvat/apps/dataset_manager/tests/test_rest_api_formats.py

```diff
@@ -0,0 +1,787 @@
+                            task_data = TaskData(task_ann.ir_data, Task.objects.get(pk=task_id))
+                            extractor = CvatTaskDataExtractor(task_data, include_images=include_images)
+                            data_from_task_after_upload = Dataset.from_extractors(extractor)
+                            compare_datasets(self, data_from_task_before_upload, data_from_task_after_upload)
+
+    # def test_api_v1_tasks_annotations_update_wrong_label(self):
+    #     test_name = self._testMethodName
+    #     dump_format_name = "CVAT for images 1.1"
+    #     upload_format_name = "CVAT 1.1"
+    #
+    #     # create task with annotations
+    #     images = self._generate_task_images(3)
+    #     task = self._create_task(tasks["main"], images)
+    #     self._create_annotations(task, dump_format_name, "default")
+    #     task_id = task["id"]
+    #     task_ann = TaskAnnotation(task_id)
+    #     task_ann.init_from_db()
+    #
+    #     # dump annotations
+    #     url = self._generate_url_dump_tasks_annotations(task_id)
+    #     data = {
+    #         "format": dump_format_name,
+    #         "action": "download",
+    #     }
+    #     with TestDir() as test_dir:
+    #         file_zip_name_before_change = osp.join(test_dir, f'{test_name}_{dump_format_name}.zip')
+    #         file_zip_name_after_change = osp.join(test_dir, f'{test_name}_{dump_format_name}_wrong_label.zip')
+    #
+    #         # download zip file
+    #         self._download_file(url, data, self.admin, file_zip_name_before_change)
+    #         self._check_downloaded_file(file_zip_name_before_change)
+    #
+    #         # remove annotations
+    #         self._remove_annotations(url, self.admin)
+    #
+    #         # extract zip
+    #         folder_name = osp.join(test_dir, f'{test_name}_{dump_format_name}')
+    #         with zipfile.ZipFile(file_zip_name_before_change, 'r') as zip_ref:
+    #             zip_ref.extractall(folder_name)
+    #
+    #         # change right label to wrong
+    #         wrong_label = "wrong_label"
+    #         tree = ET.parse(osp.join(folder_name, 'annotations.xml'))
+    #         root = tree.getroot()
+    #         element = root.find("./image[@id='0']/box[@label='car']")
+    #         element.attrib["label"] = wrong_label
+    #         tree.write(osp.join(folder_name, 'annotations.xml'))
+    #         with zipfile.ZipFile(file_zip_name_after_change, 'w') as zip_ref:
+    #             zip_ref.write(osp.join(folder_name, 'annotations.xml'), 'annotations.xml')
+    #
+    #         # upload annotations
+    #         url_upload = self._generate_url_upload_tasks_annotations(task_id, upload_format_name)
+    #         with open(file_zip_name_after_change, 'rb') as binary_file:
+    #             with self.assertRaisesRegex(ValueError, f"Label '{wrong_label}' is not registered for this task"):
+    #                 response = self._upload_file(url_upload, binary_file, self.admin)
```

Comment:
You probably shouldn't try to catch the exception here, since it was handled earlier.